### PR TITLE
Pstrunc/3156 role assignment UI

### PR DESCRIPTION
**Pull request by peter-strunc**
https://redmine.czechidm.com/issues/3156

### CODE REVIEW

File: Realization/backend/app/src/main/resources/application.properties [hunk 1]
```diff
@@ -431,7 +431,7 @@ idm.pub.app.show.identity.formProjection.default=true
 # If you want to configure rendered columns for all tables generalized from identity table (e.g. on role or tree node detail), 
 # you can use FE configuration https://wiki.czechidm.com/devel/documentation/application_configuration/dev/frontend
 idm.pub.app.show.identity.table.columns=username, lastName, firstName, externalCode, email, state, description
-idm.pub.app.show.identityRole.table.columns=role, roleAttributes, environment, identityContract, contractPosition, validFrom, validTill, directRole, automaticRole, incompatibleRoles
+idm.pub.app.show.identityRole.table.columns=role, roleAttributes, environment, owner, contractPosition, validFrom, validTill, directRole, automaticRole, incompatibleRoles
```
Comment (tomdoischer) [inline Realization/backend/app/src/main/resources/application.properties:434]: This needs to be in the changelog and documentation and I'm very worried we will forget it. :laughing:

File: Realization/frontend/czechidm-core/src/components/advanced/Filter/FilterButtons.js [hunk 2]
```diff
@@ -25,6 +25,7 @@ export default class FilterButtons extends Basic.AbstractContextComponent {
 
   _useFilter(event) {
     const { useFilter } = this.props;
+    console.log("wwww",useFilter)
```
Comment (tomdoischer) [inline Realization/frontend/czechidm-core/src/components/advanced/Filter/FilterButtons.js:28]: Remove this pls.

File: Realization/frontend/czechidm-core/src/content/requestrole/RequestIdentityRoleTable.js [hunk 3]
```diff
@@ -500,7 +502,7 @@ export class RequestIdentityRoleTable extends Advanced.AbstractTableContent {
     if (identityId && !isAccount) {
       forceSearchParameters = forceSearchParameters.setFilter('identityId', identityId);
       forceSearchParameters = forceSearchParameters.setFilter('accountId', null);
-      forceSearchParameters = forceSearchParameters.setFilter('ownerType', null);
+      //forceSearchParameters = forceSearchParameters.setFilter('ownerType', null);
```
Comment (tomdoischer) [inline Realization/frontend/czechidm-core/src/content/requestrole/RequestIdentityRoleTable.js:505]: Remove this commented-out code.

File: Realization/frontend/czechidm-core/src/content/requestrole/RoleConceptDetail.js [hunk 4]
```diff
@@ -417,6 +377,58 @@ export class RoleConceptDetail extends Basic.AbstractContent {
...
+                          ref={component.ownerType}
+                          manager={managerInstance}
+                          forceSearchParameters={forceParams}
+                          defaultSearchParameters={
+                            new SearchParameters().clearSort()
+                          }
+                          pageSize={100}
+                          //label={this.i18n('entity.IdentityRole.identityContract.label')}
+                          //placeholder={this.i18n('entity.IdentityRole.identityContract.placeholder')}
+                          //helpBlock={this.i18n('entity.IdentityRole.identityContract.help')}
```
Comment (tomdoischer) [inline Realization/frontend/czechidm-core/src/content/requestrole/RoleConceptDetail.js:415]: Should it be used or not?

File: Realization/frontend/czechidm-core/src/services/ComponentService.js [hunk 5]
```diff
@@ -193,6 +193,11 @@ export default class ComponentService {
     return this.getComponentByAttributeValue(ComponentService.ROLE_CONCEPT_MANAGER, selectedOwnerType, "ownerType");
   }
 
+  getApplicantInfoComponent(applicantType) {
+    console.log("uuuu applicantType", applicantType)
```
Comment (tomdoischer) [inline Realization/frontend/czechidm-core/src/services/ComponentService.js:197]: Remove this pls.

**Review by tomdoischer — CHANGES_REQUESTED** [review]
- Flyway scripts need some love. There are duplicates.
- Flyway scripts are not complete. ERROR: column "applicant" of relation "idm_role_request" does not exist
- Cannot assign role to a contract with error: Caused by: org.postgresql.util.PSQLException: ERROR: null value in column "applicant_id" violates not-null constraint
- Role assignment is missing in the wizard for account creation. Not sure it is a part of this ticket but there is no other ticket for it.
 
I didn't get further.

File: Realization/backend/app/src/main/resources/application.properties (hunk 1 above)
Comment (peter-strunc) [reply Realization/backend/app/src/main/resources/application.properties:434]: added mention of this to changelog

**Review by tomdoischer — CHANGES_REQUESTED** [review]
Mostly works quite well, excellent job.

Some issues.
- Cannot assign roles with bulk action on identities.
- On account detail, tab Roles, when clicking Request to change roles, a white screen appears.
- Info component for an account has a link that doesn't get to the account detail but account table.
Screenshot from 2022-12-15 09-40-06

- Localization is strange in role request detail:
Screenshot from 2022-12-15 09-35-59

**Review by tomdoischer — APPROVED** [review]

**Pull request merged**